lib/overcommit/hook/pre_commit/php_stan.rb

```diff
@@ -0,0 +1,30 @@
+module Overcommit::Hook::PreCommit
+  # Runs `phpstan` against any modified PHP files.
+  # For running `phpstan` with Laravel, it requires setup with `ide_helper`.
+  #
+  # References:
+  # https://github.com/phpstan/phpstan/issues/239
+  # https://gist.github.com/edmondscommerce/89695c9cd2584fefdf540fb1c528d2c2
+  class PhpStan < Base
+    MESSAGE_REGEX = /^(?<file>.+)\:(?<line>\d+)\:(?<message>.+)/
+
+    def run
+      messages = []
+
+      applicable_files.each do |file|
+        result = execute(command, args: [file])
+        output = result.stdout.chomp
```

Comment:
You don't need to assign this variable unless the exit status was successful, so perhaps only do this if `result.success?`

Reply:
I'll give it a try.